2433_Find_The_Original_Array_of_Prefix_Xor/try_1.cpp

```diff
@@ -0,0 +1,39 @@
+#include<bitset>
+
+using namespace std;
+
+class Solution {
+private:
+    int helper(int a, int b) {
+        // a ^ x = b
+        bitset<32> x;
+        bitset<32> bit_a(a);
+        bitset<32> bit_b(b);
+
+        for (int i=0; i<32; i++) {
+            int d1 = bit_a[i], d2 = bit_b[i];
+            
+            if (d2 == 1) {
+                x[i] = d1 == 0 ? 1 : 0;
+            } else {
+                x[i] = d1;
+            }
+
+        }
+        return (int) x.to_ulong();
+    }
+    
+public:
+    vector<int> findArray(vector<int>& pref) {
+        int n = pref.size();
+        vector<int> ans;
+        ans.push_back(pref[0]);
+        
+        for (int i=1; i < n; i++) {
+            ans.push_back(helper(pref[i-1], pref[i]));
```

Comment:
```suggestion
            ans.push_back(pref[i-1]^pref[i]);
```
寫完Truth table發現helper做的就是Xor 那可以用「^」來達成。

Reply:
是沒錯，不過目前的這些寫法是當下我跟阿寶一起討論出來的，我只是想把當下的做法翻譯成c++而已～

Reply:
了解!